### PR TITLE
Added support for maxWidth and minWidth options for regions

### PR DESCRIPTION
**Pull request by gapipro**

### CODE REVIEW

**Comment by katspaugh** [comment]
Why do you want them? What's the use case?

**Comment by gapipro** [comment]
When you are using wavesurfer for something like audio trimming. Region indicates are that will remain. And in that case you can limit user to minimum and maximum length of trimmed audio.

File: plugin/wavesurfer.regions.js [hunk 1]
```diff
@@ -89,6 +89,9 @@ WaveSurfer.Region = {
         this.color = params.color || 'rgba(0, 0, 0, 0.1)';
         this.data = params.data || {};
 
+        this.maxWidth = params.maxWidth === undefined ? null : Number(params.maxWidth);
+        this.minWidth = params.minWidth === undefined ? null : Number(params.minWidth);
```
Comment (katspaugh) [inline plugin/wavesurfer.regions.js]: * Rename `*Width` to `*Length`.
 * Just `this.*Length = params.*Length` would be enough.

File: plugin/wavesurfer.regions.js [hunk 2]
```diff
@@ -210,6 +220,15 @@ WaveSurfer.Region = {
           this.end = dur;
           this.start = dur - (this.end - this.start);
         }
+
+        if (this.maxWidth && this.end - this.start > this.maxWidth){
```
Comment (katspaugh) [inline plugin/wavesurfer.regions.js]:
```
if (this.minLength != null) {
    this.end = Math.max(this.start + this.minLength, this.end);
}

if (this.maxLength != null) {
    this.end = Math.min(this.start + this.maxLength, this.end);
}
```

**Comment by katspaugh** [comment]
Merged, thanks!

**Pull request merged**